antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -1050,3 +1051,35 @@ def initialize_additional_data(self, variant_study: VariantStudy) -> bool:
                 exc_info=e,
             )
             return False
+
+    def clear_all_snapshots(self, limit: int, params: t.Optional[RequestParameters] = None) -> None:
+        """
+        Clear all variant snapshots older than `limit` (in hours).
+        Only available for admin users.
+        `limit` must be a positive integer.
+        Raises a UserHasNotPermissionError if the user is not administrator
+        Args:
+            limit (integer): number of hours
+            params: request parameters used to identify the user status
+        Returns: None
+
+        Raises:
+            UserHasNotPermissionError
+            HTTPException
+        """
+        if limit < 0:
+            raise HTTPException(status_code=400, detail=f"Limit cannot be negative (limit={limit})")
```

Comment:
The Exception should be a `422` instead of `400`. You can look for a suitable Exception in `antarest/core/exceptions.py`.
If you think none of the `422` is ok for this case, you might create your own.

Reply:
Note that it's also possible to use a `Query` annotation on the limit argument, with "ge=0" argument to specify that it should be greater or equal to 0.

Reply:
You don't need `Query` it will create a duplicate doc for your `argument`.
What I meant is that the type of your `HTTPSExpection` should be a `422` not `400`.